core/iwasm/fast-jit/jit_regalloc.c

```diff
@@ -247,12 +247,9 @@ rc_alloc_spill_slot(RegallocContext *rc, JitReg vreg)
 
 found:
     /* Now, i is the first slot for vreg.  */
-#if 0 /* TODO: check the spill */
-  if (rc->cc->interp_frame_size + (i + stride) * 4
-      > rc->cc->total_frame_size)
-    /* No frame space for the spill area.  */
-    return 0;
-#endif
+    if ((i + stride) * 4 > rc->cc->spill_cache_size)
```

Comment:
not sure about it. The original logic is to use `rc->spill_slots` as memory. If tend to use `spill_cache_size` in `WASMInterpFrame`, maybe we should also modify the other

Reply:
Here the original logic is to check whether there is enough free space in the spill area:
```C
  if (rc->cc->interp_frame_size + (i + stride) * 4
      > rc->cc->total_frame_size)
    /* No frame space for the spill area.  */
    return 0;
```
The spill area resides in the end of the stack frame, from cc->interp_frame_size to cc->total_frame_size:
```C
/**
 * Calculate the size of interpreter area of frame of a method.
 *
 * @param all_cell_num number of all cells including local variables
 * and the working stack slots
 *
 * @return the size of interpreter area of the frame
 */
static inline unsigned
jeff_interp_interp_frame_size (unsigned all_cell_num)
{
  return align_uint (offsetof (JeffInterpFrame, lp) + all_cell_num * 5, 4);
}

/**
 * Calculate the total frame size of a method, including the Java
 * frame and the preserved space for JITed code.
 *
 * @param all_cell_num number of all cells including local variables
 * and the working stack slots
 *
 * @return the total frame size of the method
 */
static inline unsigned
jeff_interp_total_frame_size (unsigned all_cell_num)
{
  /* TODO: reduce the size after optimizing the regalloc of JIT.  */
  return offsetof (JeffInterpFrame, lp) + all_cell_num * 8 + 4 * 16;
}
```
In current design, we reserve a space inside stack frame but not in the end for the register allocator to spill. The start off set is cc->spill_cache_offset, and the size is cc->spill_cache_size. My understanding is that the logic is similar to the logic of original code.

Reply:
But seems like in `rc_alloc_spill_slot`, the search part, `for (i = 0; i < rc->spill_slot_num; i += stride)` is looking for enough space for different values, accroding to `get_reg_stride`. Every `JitReg` is a uint32, why using `stride` instead of a fixed size?